embedded-graphics/src/image/mod.rs

```diff
@@ -1,16 +1,163 @@
 //! Image object
 
+use crate::coord::Coord;
 use crate::drawable::Dimensions;
+use crate::drawable::Drawable;
+use crate::pixelcolor::PixelColor;
+use crate::transform::Transform;
+use crate::unsignedcoord::{ToSigned, UnsignedCoord};
+use core::marker::PhantomData;
 
 mod image16bpp;
 mod image1bpp;
 mod image8bpp;
 mod image_bmp;
 
-/// Raw image trait
-pub trait Image<'a>: Dimensions {
+/// Raw image type
+pub trait ImageType {}
+
+/// Raw image
+#[derive(Debug)]
+pub struct Image<'a, C, T>
+where
+    C: PixelColor,
+    T: ImageType,
+{
+    /// Image width in pixels
+    width: u32,
+
+    /// Image height in pixels
+    height: u32,
+
+    /// Image data, 1 bit per byte, 1 byte per 8 horizontal pixels
```

Comment:
This isn't correct any more as the data in the slice will be interpreted differently by the different *BPP image types. Perhaps:

```suggestion
    /// Image data, packed as dictated by image type `T` (`Image1BPP` uses 8 bits per pixel, etc)
```